connect/transport_http.go

```diff
@@ -30,8 +32,19 @@ type errorer interface {
 }
 
 func decodeRequest(ctx context.Context, r *http.Request) (interface{}, error) {
-	var req mdmConnectRequest
-	err := plist.NewDecoder(r.Body).Decode(&req)
+	var res mdm.Response
+
+	body, err := ioutil.ReadAll(r.Body)
```

Comment:
reading the whole body into memory can be a security/DDOS problem. 
Change this line to the following
```
body, err := ioutil.ReadAll(io.LimitReader( r.Body,10000))
```

LimitReader: https://golang.org/pkg/io/#LimitReader

Reply:
I agree about the DDOS concern, but I think this is not the correct location to handle this. It should be done in a far earlier middleware.

For example, the `/connect` endpoint has a preceding middleware `mdmAuthSignMessageMiddleware` to check signatures and gets the body in the same way: `bodyBuf, err := ioutil.ReadAll(r.Body)`

Running 30MB through the hashing algorithm will probably have more of a DOS impact than the plist parsing would.

Reply:
Also, 10KB may be fairly constrained for some bodies, like the application list. Typically in other projects I go with 1MB for API endpoints that are not file uploads of some kind.